### ISSUE TITLE
Validate regular expressions

### ISSUE THREAD

**Issue by Danghor** [JosefPihrt/Roslynator#533]
**Product and Version Used**: 
Roslynator 2017 (2.1.0)

**Steps to Reproduce**:
Add the following line into a `C#` method:

```csharp
Regex.Match("my text", @"\pXXX");
```

**Actual Behavior**:
No diagnostic is created.

**Expected Behavior**:
Since the second parameter (the `pattern`) is invalid and throws an exception at runtime, an analyzer should detect this error.

This is the sample analyzer used in these two MSDN articles.
- https://msdn.microsoft.com/magazine/dn879356.aspx
- https://msdn.microsoft.com/magazine/dn904670.aspx
It seems useful and I think Roslynator should implement it.

**Comment by Danghor** [comment]
In Visual Studio 2019, this is already covered by RE0001.
